### PR TITLE
removed Groovy support from tests

### PR DESCRIPTION
**Pull request by kaikreuzer**
(helps making the TP simpler)

Signed-off-by: Kai Kreuzer <kai@openhab.org>

### CODE REVIEW

**Comment by kaikreuzer** [comment]
Done.

**Comment by maggu2810** [comment]
That's interesting that the `PR-openHAB-Core` succeeded and the `continuous-integration/travis-ci/pr` failed because of:

```
[ERROR] Tests not run because project has errors
[ERROR] Error   : Cannot find /error/groovy;version=[2.5.5,2.5.6) Not found in [openHAB Core :: Integration Tests :: Tests [org.openhab.core,org.openhab.core.test,org.openhab.core.tests r/w=false], bnd-cache [/home/travis/build/openhab/openhab-core/itests/org.openhab.core.tests/target/test/integration-test/cnf/cache/4.1.0/bnd-cache r/w=false], runtime [/home/travis/build/openhab/openhab-core/itests/org.openhab.core.tests/target/test/integration-test/cnf/cache/4.1.0/runtime r/w=false], test [/home/travis/build/openhab/openhab-core/itests/org.openhab.core.tests/target/test/integration-test/cnf/cache/4.1.0/test r/w=false], self [/home/travis/build/openhab/openhab-core/itests/org.openhab.core.tests/target/test/integration-test/cnf/cache/4.1.0/self r/w=false]]
[ERROR] Error   : groovy;version=[2.5.5,2.5.6) Not found in [openHAB Core :: Integration Tests :: Tests [org.openhab.core,org.openhab.core.test,org.openhab.core.tests r/w=false], bnd-cache [/home/travis/build/openhab/openhab-core/itests/org.openhab.core.tests/target/test/integration-test/cnf/cache/4.1.0/bnd-cache r/w=false], runtime [/home/travis/build/openhab/openhab-core/itests/org.openhab.core.tests/target/test/integration-test/cnf/cache/4.1.0/runtime r/w=false], test [/home/travis/build/openhab/openhab-core/itests/org.openhab.core.tests/target/test/integration-test/cnf/cache/4.1.0/test r/w=false], self [/home/travis/build/openhab/openhab-core/itests/org.openhab.core.tests/target/test/integration-test/cnf/cache/4.1.0/self r/w=false]]
```

The integration tests define which stuff is required for execution. See: https://github.com/openhab/openhab-core/blob/76dd62623ff35298f068e520dd994ac6f57f4a80/itests/org.openhab.core.tests/integration-test.bndrun#L24

The resolution is done once (using the Eclipse IDE or a maven goal) and then the bundles that are used at runtime are fixed.

This is much better then the previous setup. There has been Eclipse launch configurations to execute the tests in the IDE and a POM based configuration to execute the tests on a Maven build.
Launch configurations are broken multiple times without being detected as long as no one use it in the IDE.

Now, the configuration is used by the IDE and the Maven build and it has been detected that the old resolution does not work anymore.

Previously the groovy bundle has been resolved that needs to available:
https://github.com/openhab/openhab-core/blob/76dd62623ff35298f068e520dd994ac6f57f4a80/itests/org.openhab.core.tests/integration-test.bndrun#L33

Now, this "runbundles" cannot be used anymore as the Groovy one is not available anymore.

We need to rerun the resolution once so Groovy will not be part of the bundle set anymore (as not requirement).

**Comment by kaikreuzer** [comment]
> We need to rerun the resolution once so Groovy will not be part of the bundle set anymore

Err, can you elaborate on what this exactly means...?

**Comment by kaikreuzer** [comment]
Ah, sorry, only saw https://github.com/kaikreuzer/openhab-core/pull/2 right now.

File: itests/org.openhab.core.tests/integration-test.bndrun [hunk 1]
```diff
@@ -46,14 +45,14 @@ Provide-Capability: osgi.service;objectClass:List<String>="org.eclipse.smarthome
...
 	org.mockito.mockito-core;version='[2.13.0,2.13.1)',\
 	org.objenesis;version='[2.6.0,2.6.1)',\
 	org.osgi.service.event;version='[1.4.0,1.4.1)',\
 	osgi.enroute.junit.wrapper;version='[4.12.0,4.12.1)',\
 	slf4j.api;version='[1.7.21,1.7.22)',\
 	tec.uom.lib.uom-lib-common;version='[1.0.2,1.0.3)',\
 	tec.uom.se;version='[1.0.8,1.0.9)',\
-	osgi.enroute.hamcrest.wrapper;version='[1.3.0,1.3.1)'
+	osgi.enroute.hamcrest.wrapper;version='[1.3.0,1.3.1)',\
+	org.openhab.core;version='[2.5.0,2.5.1)',\
```
Comment (martinvw) [inline itests/org.openhab.core.tests/integration-test.bndrun]: Will this break after 2.5.1? Why

Comment (maggu2810) [reply itests/org.openhab.core.tests/integration-test.bndrun]: The version range has been chosen by the tooling. So perhaps question about should be discussed upstream (with the tool developers) or by reading their code base:
https://github.com/bndtools/bnd/tree/master/maven/bnd-resolver-maven-plugin

**Comment by maggu2810** [comment]
This PR contains a commit of myself. Perhaps I shouldn't have merged it.

**Pull request merged**